src/window_selectable.cpp

```diff
@@ -123,8 +123,13 @@ void Window_Selectable::UpdateArrows() {
 }
 
 // Update
-void Window_Selectable::Update() {
-	Window_Base::Update();
+void Window_Selectable::Update(bool block_system_change) {
+	if (block_system_change) {
+		Window::Update();
+		Window_Base::UpdateMovement();
```

Comment:
This is a gross hack, especially in #2451 where these 2 lines just appear and is totally not obvious why.

You should either:
1. Refactor the update routines so that all child classes must explicitly call an `UpdateSystemGraphic()` in their update routines.
2. Add some member variable flag or virtual function which says whether or not to update the system graphic and adjust it / call it as needed.

Basically lets make "should we update the system graphic?" a first class concept of the `Window` classes